distro-build/nixos.sh

```diff
@@ -0,0 +1,53 @@
+dist_name="NixOS"
+dist_version="22.05"
+
+bootstrap_distribution() {
+	for arch in aarch64 i686 x86_64; do
+		curl --fail --location \
+			--output "${WORKDIR}/nixos-system-${arch}-linux.tar.xz" \
+			"https://hydra.nixos.org/job/nixos/release-${dist_version}/nixos.containerTarball.${arch}-linux/latest/download/1"
+
+		sudo mkdir -m 755 "${WORKDIR}/nixos-$(translate_arch "$arch")"
+		sudo tar -Jxp \
+			-f "${WORKDIR}/nixos-system-${arch}-linux.tar.xz" \
+			-C "${WORKDIR}/nixos-$(translate_arch "$arch")"
+
+		system_dir="$(find "${WORKDIR}/nixos-$(translate_arch "$arch")/nix/store" -name "*nixos-system-nixos-${dist_version}.*")"
+		libgcc_dir="$(find "${WORKDIR}/nixos-$(translate_arch "$arch")/nix/store" -maxdepth 1 -name "*gcc*")/lib"
+
+		cat <<- EOF | sudo unshare -mpf bash -e -
+		mount --bind /dev "${WORKDIR}/nixos-$(translate_arch "$arch")/dev"
+		mount --bind /proc "${WORKDIR}/nixos-$(translate_arch "$arch")/proc"
+		mount --bind /sys "${WORKDIR}/nixos-$(translate_arch "$arch")/sys"
+		mkdir "${WORKDIR}/nixos-$(translate_arch "$arch")/etc"
+		touch "${WORKDIR}/nixos-$(translate_arch "$arch")/etc/profile"
+		mkdir "${WORKDIR}/nixos-$(translate_arch "$arch")/etc/profile.d"
+		mkdir -p "${WORKDIR}/nixos-$(translate_arch "$arch")/usr/lib"
+		ln -s "$libgcc_dir/libgcc_s.so.1" "${WORKDIR}/nixos-$(translate_arch "$arch")/usr/lib/libgcc_s.so.1"
+		chroot "${WORKDIR}/nixos-$(translate_arch "$arch")" ${system_dir#"${WORKDIR}/nixos-$(translate_arch "$arch")"}/activate
+		ln -s "$system_dir/sw/bin/su" "${WORKDIR}/nixos-$(translate_arch "$arch")/bin/su"
```

Comment:
Symlink to a system symlink doesn't seem great. If we do a `nixos-rebuild` and then a GC, then this symlink will become stale.

This comes from the tiny `shadow.su` package. If we readlink on that then we can put a gcroot in to stop that single package from being garbage collected. I'm not sure _when_ we'd put in a gcroot, but we could, in theory.